thirdparty/build.gradle

```diff
@@ -23,10 +23,22 @@ dependencies {
     shaded("commons-httpclient:commons-httpclient:3.0.1")
     shaded("org.codehaus.jackson:jackson-mapper-asl:${project.ext.jacksonVersion}")
     shaded("org.codehaus.jackson:jackson-core-asl:${project.ext.jacksonVersion}")
+    shaded("com.google.guava:guava:${project.ext.googleGuavaVersion}")
+    shaded("software.amazon.awssdk:sdk-core:${project.ext.awsSdkVersion}")
```

Comment:
Is there any guidance which package should be putted into shadow jar? I think depend on [aws-java-sdk-bundle ](https://aws.amazon.com/blogs/developer/java-sdk-bundle/) is [more generic approach](https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/index.html#Getting_Started).

Reply:
I think we would need to package the dependencies in the jar either way, looks like with shadow jar we control the exact dependencies? I could be wrong, @reta any input?

Reply:
aws-java-sdk-bundle is the shadowjar, it's the generic runtiime depedency for spark. opensearch-hadoop do not need to shaowjar aws-java-sdk-bundle.

Reply:
@penghuo I think the intent here was to have `opensearch-hadoop/mr` JAR self-sufficient: the AWS dependency are not bundled with Hadoop or/and Spark, if we don't shade - we have to make them available to these runtimes somehow (@harshavamsi I think you have an exception shared on the pull request). I haven't looked into `aws-java-sdk-bundle`, will do that tomorrow.

Reply:
@reta totally agree. aws-java-sdk-bundle is the shaded jar. Hadoop s3 depend on it https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/index.html#Getting_Started .

Reply:
add sample code to explain it. https://github.com/penghuo/opensearch-hadoop/commit/098cfd9865e45f508fc5bc6d4c93b254ee46d49f#diff-4f0cebafafd8b671aaf175130520fa2953152f3637bedfaa9191441462f6d93c

```
import org.apache.spark.sql.SQLContext
import org.apache.spark.sql.SQLContext._
import org.opensearch.spark.sql._
val sql = new SQLContext(sc)
                         
val options = Map("pushdown" -> "true",     
"opensearch.nodes" -> "https://search-mytest-00001-huifxbwhdgfy7goat3mietmxne.us-west-2.es.amazonaws.com", 
"opensearch.aws.sigv4.enabled" -> "true",
"opensearch.aws.sigv4.region" -> "us-west-2",
"opensearch.nodes.resolve.hostname" -> "false",
"opensearch.net.ssl" -> "true")                         

val es = sql.esDF("accounts", options)
```

we only need to add aws-java-sdk-bundle to spark/jars, no shaded jar required.

Reply:
@penghuo thanks! So what we could do is 2 things:
 - document that  `aws-java-sdk-bundle` has to be added to runtime classpath (fe spark/jars)
 - refactor the code so the AwsSig is becoming optional, in this case if it is not used, the `aws-java-sdk-bundle` won't be required

Reply:
You have to copy it manually

Reply:
@reta I add `implementation 'com.amazonaws:aws-java-sdk-bundle:1.12.397'` [here](https://github.com/opensearch-project/opensearch-hadoop/pull/48/files#diff-e1955902ceda6d0d39c7891161eb56eadfde6d1a78897908ab7043af09105258R109) but it does not seem to help with the `NoClassDef` error

Reply:
I mean you have to manually copy the com.amazonaws:aws-java-sdk-bundle:1.12.397 jar to spark/jars, as per @penghuo comment https://github.com/opensearch-project/opensearch-hadoop/pull/48#discussion_r1094103160

Reply:
@harshavamsi summarize my idea in https://github.com/penghuo/opensearch-hadoop/commit/0feda06f0e9c36834d3cd3cf150cb24d111d3e09

it is not production code, only for explain the idea.

## Test Setup
* copy the follow jars to spark/jar folder
```
 ~/spark-3.2.1-bin-hadoop3.2  ls jars/aws-java-sdk-bundle-1.11.375.jar
jars/aws-java-sdk-bundle-1.11.375.jar
 ~/spark-3.2.1-bin-hadoop3.2  ls jars/opensearch-spark-30_2.12-3.0.0-SNAPSHOT.jar
jars/opensearch-spark-30_2.12-3.0.0-SNAPSHOT.jar
```
* configure AWS credentials 

## Test code

* Write to AOS

```
import org.apache.spark.sql.SQLContext
import org.apache.spark.sql.SQLContext._
import org.opensearch.spark.sql._

val sql = new SQLContext(sc)

val accountsRead = sql.read.json("/Users/penghuo/release/maximus/data/accounts.json")
val options = Map("pushdown" -> "true",     
"opensearch.nodes" -> "https://aos-2.3-domain", 
"opensearch.aws.sigv4.enabled" -> "true",
"opensearch.aws.sigv4.region" -> "us-west-2",
"opensearch.nodes.resolve.hostname" -> "false",
"opensearch.nodes.wan.only" -> "true",
"opensearch.net.ssl" -> "true")   
accountsRead.saveToOpenSearch("accounts-00001", options)
```

* Load from AOS
```
import org.apache.spark.sql.SQLContext
import org.apache.spark.sql.SQLContext._
import org.opensearch.spark.sql._
val sql = new SQLContext(sc)
                         
val options = Map("pushdown" -> "true",     
"opensearch.nodes" -> "https://aos-2.3-domain", 
"opensearch.aws.sigv4.enabled" -> "true",
"opensearch.aws.sigv4.region" -> "us-west-2",
"opensearch.nodes.resolve.hostname" -> "false",
"opensearch.nodes.wan.only" -> "true",
"opensearch.net.ssl" -> "true")                         

val es = sql.esDF("accounts", options)
```